staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/generic/webhook.go

```diff
@@ -207,6 +222,43 @@ func (a *Webhook) ShouldCallHook(h webhook.WebhookAccessor, attr admission.Attri
 		return nil, matchObjErr
 	}
 
+	versionedAttr, err := admission.NewVersionedAttributes(attr, invocation.Kind, o)
+	invocation.VersionedAttr = versionedAttr
+
+	matchConditions := h.GetMatchConditions()
+	if len(matchConditions) > 0 {
+		expressions := make([]cel.ExpressionAccessor, len(matchConditions))
+		for i, matchCondition := range matchConditions {
+			expressions[i] = &MatchCondition{
+				Name:            matchCondition.Name,
+				Expression:      matchCondition.Expression,
+				WebhookStepType: h.GetWebhookStepType(),
+			}
+		}
+
+		matcher := a.newMatcher(a.filterCompiler.Compile(
+			expressions,
+			cel.OptionalVariableDeclarations{
+				HasParams:     false,
+				HasAuthorizer: true,
+			},
+			celconfig.PerCallLimit,
+		), a.authorizer)
```

Comment:
this compilation should happen once per revision of the admission config object, only Match should be called inside this function

Reply:
follow the example of mutatingWebhookAccessor#GetRESTClient building the expensive thing once in the accessor, storing the item and error in the accessor, returning those here

also add a graduation TODO to resolve the fact that we rebuild ALL items whenever ANY config changes in NewMutatingWebhookConfigurationManager and NewValidatingWebhookConfigurationManager ... now that we're doing CEL compilation, we probably want to avoid that

Reply:
done